### ISSUE TITLE
autokey-gtk crashes after update to Python 3.7 (python-atspi problem)

### ISSUE THREAD

**Issue by dark-saber** [autokey/autokey#173]
## Classification:

Crash

## Reproducibility:

Always

## Summary

After update to Python 3.7 on Arch Linux, autokey-gtk crashes on startup if python-atspi is installed. It starts fine if python-atspi is not installed, autokey-qt is also unaffected.

## Steps to Reproduce (if applicable)

- Update to Python 3.7
- Have python-atspi installed
- Run autokey-gtk

## Expected Results

autokey-gtk should start normally

## Actual Results

- Instead, autokey-gtk crashes immediately
```
autokey-gtk --verbose

Traceback (most recent call last):
  File "/usr/bin/autokey-gtk", line 11, in <module>
    load_entry_point('autokey==0.95.2', 'console_scripts', 'autokey-gtk')()
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 476, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2700, in load_entry_point
    return ep.load()
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2318, in load
    return self.resolve()
  File "/usr/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2324, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python3.7/site-packages/autokey/gtkui/__main__.py", line 4, in <module>
    from autokey.gtkapp import Application
  File "/usr/lib/python3.7/site-packages/autokey/gtkapp.py", line 44, in <module>
    from autokey import service, monitor
  File "/usr/lib/python3.7/site-packages/autokey/service.py", line 25, in <module>
    from autokey.iomediator.key import Key
  File "/usr/lib/python3.7/site-packages/autokey/iomediator/__init__.py", line 20, in <module>
    from ._windowgrabber import WindowGrabber
  File "/usr/lib/python3.7/site-packages/autokey/iomediator/_windowgrabber.py", line 21, in <module>
    from ._iomediator import IoMediator
  File "/usr/lib/python3.7/site-packages/autokey/iomediator/_iomediator.py", line 7, in <module>
    from ..interface import XRecordInterface, AtSpiInterface
  File "/usr/lib/python3.7/site-packages/autokey/interface.py", line 54, in <module>
    import pyatspi
  File "/usr/lib/python3.7/site-packages/pyatspi/__init__.py", line 21, in <module>
    from pyatspi.Accessibility import *
  File "/usr/lib/python3.7/site-packages/pyatspi/Accessibility.py", line 17, in <module>
    from pyatspi.registry import *
  File "/usr/lib/python3.7/site-packages/pyatspi/registry.py", line 114
    self.async = False  # not fully supported yet
             ^
SyntaxError: invalid syntax
```

## Version

AutoKey version: 0.95.2

Other package versions: python 3.7.0; python-atspi 2.26.0

Used GUI (Gtk, Qt, or both): Gtk

Installed via: Arch Linux package manager (AUR)

Distro: Arch Linux

## Notes

The current workaround is to remove python-atspi, but it may be needed for other applications.

**Comment by luziferius** [comment]
Thank you for pinging us. This seems to be a bug in pyatspi.

You are the AUR maintainer, right?

`pyatspi` is imported here:
https://github.com/autokey/autokey/blob/abf33455a2d2c6ae2ae848c6621f19564392340f/lib/autokey/interface.py#L52-L59
The presence is optional and meant to be an alternative choice, if XRecordInterface fails for the user (which is unlikely).

You can just additionally catch `SyntaxError` like this and autokey-gtk should continue to work: 
```
except SyntaxError:  # pyatspi 2.26 fails when used with Python 3.7
    HAS_ATSPI = False
```
You can then (temporarily?) drop the dependency.

By the way:
- Can you update the upstream URL to https://github.com/autokey/autokey? The old link does redirect, but it is nonetheless outdated.
- You may want to depend on `kdialog`, because the Qt GUI uses KDialog instead of zenity for better integration with KDE based desktops.

Edit: This is the culprit:

Quote from https://docs.python.org/3/whatsnew/3.7.html:

Backwards incompatible syntax changes:
 - async and await are now reserved keywords.

**Comment by feluxe** [comment]
>  This seems to be a bug in pyatspi.

Looks like it was fixed recently: https://gitlab.gnome.org/GNOME/pyatspi2/issues/1

**Comment by luziferius** [comment]
@feluxe Thanks for pinging :+1: 

@dark-saber The crash fix workaround is included in the just released 0.95.3, so you can drop the `atspi.patch`. Additionally, this will conflict with the `requires.patch`. I’m still not sure how to properly handle the problematic `dbus-python`, so I haven’t removed that dependency.
But the patch for `python-xlib` is included in this release, here: https://github.com/autokey/autokey/commit/d6f0c3ec8402601f087e101f937db31e6aa16475.

I’ll close this issue for now.
Feel free to re-open if things still need some cleanups.
